File: backend/src/main/kotlin/mymoney/domain/model/Enums.kt
package mymoney.domain.model

enum class TransactionType { INCOME, EXPENSE, TRANSFER }

enum class CategoryType { INCOME, EXPENSE }

enum class FamilyRole { OWNER, MEMBER }

enum class BudgetPeriodType { WEEK, MONTH, YEAR }

enum class DebtDirection { I_OWE, OWED_TO_ME }

enum class DebtStatus { OPEN, CLOSED }

File: mobile/lib/domain/model/enums.dart
enum CategoryType { income, expense }

enum TransactionType { income, expense, transfer }

extension CategoryTypeX on CategoryType {
  String get code => switch (this) {
        CategoryType.income => 'INCOME',
        CategoryType.expense => 'EXPENSE',
      };
}

extension TransactionTypeX on TransactionType {
  String get code => switch (this) {
        TransactionType.income => 'INCOME',
        TransactionType.expense => 'EXPENSE',
        TransactionType.transfer => 'TRANSFER',
      };
}
